app/services/foreman/client_certificate.rb

```diff
@@ -0,0 +1,63 @@
+module Foreman
+  class ClientCertificate
+    delegate :logger, to: :Rails
+    attr_reader :request
+
+    def initialize(request:)
+      @request = request
+    end
+
+    def raw_cert_available?
+      @raw_cert_available ||= request.ssl? && request.env.has_key?(certificate_env_key) && raw_data.present? && raw_data != '(null)'
```

Comment:
no need for the `has_key?` check here, if the key isn't present `raw_data` will return `nil` and `raw_data.present?` will still be false.
```suggestion
      @raw_cert_available ||= request.ssl? && raw_data.present? && raw_data != '(null)'
```